diff -ruN libiconv-1.19/srclib/fseterr.c libiconv-1.19.aros/srclib/fseterr.c
--- libiconv-1.19/srclib/fseterr.c	2026-03-06 13:36:51.000000000 +0000
+++ libiconv-1.19.aros/srclib/fseterr.c	2026-07-26 18:47:41.959953087 +0100
@@ -53,7 +53,7 @@
 #elif defined EPLAN9                /* Plan9 */
   if (fp->state != 0 /* CLOSED */)
     fp->state = 5 /* ERR */;
-#elif 0                             /* unknown  */
+#elif defined __AROS__
   /* Portable fallback, based on an idea by Rich Felker.
      Wow! 6 system calls for something that is just a bit operation!
      Not activated on any system, because there is no way to repair FP when
diff -ruN libiconv-1.19/srclib/getprogname.c libiconv-1.19.aros/srclib/getprogname.c
--- libiconv-1.19/srclib/getprogname.c	2026-03-06 13:36:51.000000000 +0000
+++ libiconv-1.19.aros/srclib/getprogname.c	2026-07-26 18:47:41.960016987 +0100
@@ -251,6 +251,8 @@
       close (fd);
     }
   return "?";
+# elif defined __AROS__
+  return "";
 # else
 #  error "getprogname module not ported to this OS"
 # endif
