diff -ruN SDL_mixer-1.2.12/build-scripts/config.sub SDL_mixer-1.2.12.aros/build-scripts/config.sub
--- SDL_mixer-1.2.12/build-scripts/config.sub	2012-01-15 22:01:05.000000000 +0000
+++ SDL_mixer-1.2.12.aros/build-scripts/config.sub	2026-07-26 19:21:47.115655369 +0100
@@ -245,6 +245,8 @@
 	# Some are omitted here because they have special meanings below.
 	1750a | 580 \
 	| a29k \
+	| aarch64 | aarch64_be \
+	| riscv | riscv32 | riscv32be | riscv64 | riscv64be \
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
 	| am33_2.0 \
@@ -326,6 +328,8 @@
 	# Recognize the basic CPU types with company name.
 	580-* \
 	| a29k-* \
+	| aarch64-* | aarch64_be-* \
+	| riscv-* | riscv32-* | riscv32be-* | riscv64-* | riscv64be-* \
 	| alpha-* | alphaev[4-8]-* | alphaev56-* | alphaev6[78]-* \
 	| alpha64-* | alpha64ev[4-8]-* | alpha64ev56-* | alpha64ev6[78]-* \
 	| alphapca5[67]-* | alpha64pca5[67]-* | arc-* \
diff -ruN SDL_mixer-1.2.12/configure SDL_mixer-1.2.12.aros/configure
--- SDL_mixer-1.2.12/configure	2012-01-15 22:01:05.000000000 +0000
+++ SDL_mixer-1.2.12.aros/configure	2026-07-26 19:21:47.115317769 +0100
@@ -12013,7 +12013,7 @@
 
   if test "x$sdl_prefix$sdl_exec_prefix" = x ; then
 
-pkg_failed=no
+pkg_failed=yes
 { echo "$as_me:$LINENO: checking for SDL" >&5
 echo $ECHO_N "checking for SDL... $ECHO_C" >&6; }
 
@@ -13576,7 +13576,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lvorbisfile  $LIBS"
+LIBS="-lvorbisfile -lvorbis -logg $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
diff -ruN SDL_mixer-1.2.12/dynamic_mod.h SDL_mixer-1.2.12.aros/dynamic_mod.h
--- SDL_mixer-1.2.12/dynamic_mod.h	2012-01-15 22:01:05.000000000 +0000
+++ SDL_mixer-1.2.12.aros/dynamic_mod.h	2026-07-26 19:21:47.102306469 +0100
@@ -30,11 +30,11 @@
 	void (*MikMod_Exit)(void);
 	CHAR* (*MikMod_InfoDriver)(void);
 	CHAR* (*MikMod_InfoLoader)(void);
-	BOOL (*MikMod_Init)(CHAR*);
+	int (*MikMod_Init)(const char*);
 	void (*MikMod_RegisterAllLoaders)(void);
 	void (*MikMod_RegisterDriver)(struct MDRIVER*);
 	int* MikMod_errno;
-	char* (*MikMod_strerror)(int);
+	const char* (*MikMod_strerror)(int);
 	BOOL (*Player_Active)(void);
 	void (*Player_Free)(MODULE*);
 	MODULE* (*Player_LoadGeneric)(MREADER*,int,BOOL);
diff -ruN SDL_mixer-1.2.12/dynamic_ogg.h SDL_mixer-1.2.12.aros/dynamic_ogg.h
--- SDL_mixer-1.2.12/dynamic_ogg.h	2012-01-15 22:01:05.000000000 +0000
+++ SDL_mixer-1.2.12.aros/dynamic_ogg.h	2026-07-26 19:21:47.102357769 +0100
@@ -31,7 +31,7 @@
 	void *handle;
 	int (*ov_clear)(OggVorbis_File *vf);
 	vorbis_info *(*ov_info)(OggVorbis_File *vf,int link);
-	int (*ov_open_callbacks)(void *datasource, OggVorbis_File *vf, char *initial, long ibytes, ov_callbacks callbacks);
+	int (*ov_open_callbacks)(void *datasource, OggVorbis_File *vf, const char *initial, long ibytes, ov_callbacks callbacks);
 	ogg_int64_t (*ov_pcm_total)(OggVorbis_File *vf,int i);
 #ifdef OGG_USE_TREMOR
 	long (*ov_read)(OggVorbis_File *vf,char *buffer,int length, int *bitstream);
diff -ruN SDL_mixer-1.2.12/Makefile.in SDL_mixer-1.2.12.aros/Makefile.in
--- SDL_mixer-1.2.12/Makefile.in	2012-01-15 22:01:04.000000000 +0000
+++ SDL_mixer-1.2.12.aros/Makefile.in	2026-07-26 19:21:47.102417769 +0100
@@ -33,7 +33,7 @@
 SDL_CFLAGS = @SDL_CFLAGS@
 SDL_LIBS = @SDL_LIBS@
 
-TARGET  = libSDL_mixer.la
+SDL_TARGET  = libSDL_mixer.la
 OBJECTS = @OBJECTS@
 VERSION_OBJECTS = @VERSION_OBJECTS@
 PLAYWAVE_OBJECTS = @PLAYWAVE_OBJECTS@
@@ -47,7 +47,7 @@
 LT_REVISION = @LT_REVISION@
 LT_LDFLAGS  = -no-undefined -rpath $(libdir) -release $(LT_RELEASE) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
-all: $(srcdir)/configure Makefile $(objects) $(objects)/$(TARGET) $(objects)/playwave$(EXE) $(objects)/playmus$(EXE)
+all: $(srcdir)/configure Makefile $(objects) $(objects)/$(SDL_TARGET) $(objects)/playwave$(EXE) $(objects)/playmus$(EXE)
 
 $(srcdir)/configure: $(srcdir)/configure.in
 	@echo "Warning, configure.in is out of date"
@@ -62,14 +62,14 @@
 
 .PHONY: all install install-hdrs install-lib install-bin uninstall uninstall-hdrs uninstall-lib uninstall-bin clean distclean dist
 
-$(objects)/$(TARGET): $(OBJECTS) $(VERSION_OBJECTS)
+$(objects)/$(SDL_TARGET): $(OBJECTS) $(VERSION_OBJECTS)
 	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJECTS) $(VERSION_OBJECTS) $(LDFLAGS) $(EXTRA_LDFLAGS) $(LT_LDFLAGS)
 
-$(objects)/playwave$(EXE): $(objects)/playwave.lo $(objects)/$(TARGET)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playwave.lo $(SDL_CFLAGS) $(SDL_LIBS) $(objects)/$(TARGET)
+$(objects)/playwave$(EXE): $(objects)/playwave.lo $(objects)/$(SDL_TARGET)
+	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playwave.lo $(LDFLAGS) $(EXTRA_LDFLAGS) $(SDL_LIBS) $(objects)/$(SDL_TARGET)
 
-$(objects)/playmus$(EXE): $(objects)/playmus.lo $(objects)/$(TARGET)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playmus.lo $(SDL_CFLAGS) $(SDL_LIBS) $(objects)/$(TARGET)
+$(objects)/playmus$(EXE): $(objects)/playmus.lo $(objects)/$(SDL_TARGET)
+	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playmus.lo $(LDFLAGS) $(EXTRA_LDFLAGS) $(SDL_LIBS) $(objects)/$(SDL_TARGET)
 
 install: all install-hdrs install-lib #install-bin
 install-hdrs:
@@ -80,9 +80,9 @@
 	done
 	$(SHELL) $(auxdir)/mkinstalldirs $(libdir)/pkgconfig
 	$(INSTALL) -m 644 SDL_mixer.pc $(libdir)/pkgconfig/
-install-lib: $(objects) $(objects)/$(TARGET)
+install-lib: $(objects) $(objects)/$(SDL_TARGET)
 	$(SHELL) $(auxdir)/mkinstalldirs $(libdir)
-	$(LIBTOOL) --mode=install $(INSTALL) $(objects)/$(TARGET) $(libdir)/$(TARGET)
+	$(LIBTOOL) --mode=install $(INSTALL) $(objects)/$(SDL_TARGET) $(libdir)/$(SDL_TARGET)
 install-bin:
 	$(SHELL) $(auxdir)/mkinstalldirs $(bindir)
 	$(LIBTOOL) --mode=install $(INSTALL) -m 755 $(objects)/playwave$(EXE) $(bindir)/playwave$(EXE)
@@ -98,7 +98,7 @@
 	rm -f $(libdir)/pkgconfig/SDL_mixer.pc
 	-rmdir $(libdir)/pkgconfig
 uninstall-lib:
-	$(LIBTOOL) --mode=uninstall rm -f $(libdir)/$(TARGET)
+	$(LIBTOOL) --mode=uninstall rm -f $(libdir)/$(SDL_TARGET)
 uninstall-bin:
 	rm -f $(bindir)/playwave$(EXE)
 	rm -f $(bindir)/playmus$(EXE)
diff -ruN SDL_mixer-1.2.12/mixer.c SDL_mixer-1.2.12.aros/mixer.c
--- SDL_mixer-1.2.12/mixer.c	2012-01-15 22:01:05.000000000 +0000
+++ SDL_mixer-1.2.12.aros/mixer.c	2026-07-26 19:21:47.102501169 +0100
@@ -538,6 +538,14 @@
 }
 
 
+static void Mix_FreeWAV(Uint32 magic, void *buf)
+{
+	if ( magic == WAVE || magic == RIFF )
+		SDL_FreeWAV(buf);
+	else
+		free(buf);
+}
+
 /*
  * !!! FIXME: Ideally, we want a Mix_LoadSample_RW(), which will handle the
  *             generic setup, then call the correct file format loader.
diff -ruN SDL_mixer-1.2.12/music_mod.c SDL_mixer-1.2.12.aros/music_mod.c
--- SDL_mixer-1.2.12/music_mod.c	2012-01-15 22:01:05.000000000 +0000
+++ SDL_mixer-1.2.12.aros/music_mod.c	2026-07-26 19:21:47.102643969 +0100
@@ -149,7 +149,7 @@
 	SDL_RWops *rw;
 } LMM_MREADER;
 
-BOOL LMM_Seek(struct MREADER *mr,long to,int dir)
+int LMM_Seek(struct MREADER *mr,long to,int dir)
 {
 	LMM_MREADER* lmmmr = (LMM_MREADER*)mr;
 	if ( dir == SEEK_SET ) {
diff -ruN SDL_mixer-1.2.12/timidity/config.h SDL_mixer-1.2.12.aros/timidity/config.h
--- SDL_mixer-1.2.12/timidity/config.h	2012-01-15 22:01:05.000000000 +0000
+++ SDL_mixer-1.2.12.aros/timidity/config.h	2026-07-26 19:21:47.102702469 +0100
@@ -165,6 +165,8 @@
 
 #if defined(__WIN32__) || defined(__OS2__)
 #define DEFAULT_PATH	"C:\\TIMIDITY"
+#elif defined(__AROS__)
+#define DEFAULT_PATH    "PROGDIR:"
 #else
 #define DEFAULT_PATH	"/etc/timidity"
 #define DEFAULT_PATH1	"/usr/share/timidity"
